### ISSUE TITLE
Also encode the x/y/z tile coordinate in the tiles?

### ISSUE THREAD

**Issue by trasch** [maplibre/maplibre-tile-spec#67]
I am honestly not sure if I missed something in the spec or if my usecase is somewhat special, but with mvt tiles I often have the problem that you can't decode them unless you also have the x/y/z tile coordinate, since coordinates are encoded relative to the tile bounding box.

So I would propose to also encode the tile coordinate(s) in the mlt tiles, which would make it easier to debug unknown tiles.

**Comment by nyurik** [comment]
Tile intentionally does not have zxy coordinates for several reasons:
* some tiles could be identical in multiple locations (e.g. oceans or parks with no other features)
* we work really hard to make the tile is as small as possible, and adding 3 additional values to each tile that are not needed for the vast majority of the cases is a pretty big waste.

**Comment by trasch** [comment]
> Tile intentionally does not have zxy coordinates for several reasons:
> 
> * some tiles could be identical in multiple locations (e.g. oceans or parks with no other features)

You are talking about deduplication of empty or equal tiles? That's indeed an interesting case.

> * we work really hard to make the tile is as small as possible, and adding 3 additional values to each tile that are not needed for the vast majority of the cases is a pretty big waste.

The additional 9 bytes (assuming no one wants more than 32 zoom levels, which might be wrong already) are pretty negligible compared to the size of the feature rich tiles that I usually work with, but I can see that they can make a difference if you expect most of your tiles to be empty or very small, yes.

But if I'm the only professional tile user who sees value in this then we can just as well close this isse as it is then evidentally an edge case...

**Comment by nyurik** [comment]
Correct, most tile packaging formats (mbtiles, pmtiles) even have special handling of the duplicate identical tiles - so they get stored only once. When I was working on OpenMapTiles, I remember seeing more than a hundred different "nearly-empty" tiles, some with hundreds of thousands of copies.

One other consideration to my above comments:  MVT is "concatenatable" -- I can take two MVT tiles and simply put them one after the other -- and it will work. There is no per-tile metadata, only per-layer, so a tile is simply a list of layer blobs. Martin tile server for example takes advantage of it by producing combined tiles on the fly.  I would like MLT to have the same feature.  Having per tile metadata would break this model.  Instead, you can easily achieve this with a custom layer that simply contains a single point feature with x and y tile coordinates as fields.

**Comment by trasch** [comment]
Concatenating tiles is also one of my use cases. But the tiles always need to be at the same tile coordinate, or you will end up with strange results. Like, you can't merge tiles from London and New York and expect something useful - or am I missing something here?

**Comment by nyurik** [comment]
Correct - you would want the tiles to have the same x/y/z -- but you cannot easily concatenate tiles if they have tile-level metadata. E.g. if a tile has this structure:  `[metadata][layerA][layerB]` -- you cannot just add another `[metadata][layerC]` to it -- parsers would have to be aware of multiple metadata blocks, making parsing more complex.
